layers/tls.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2012 Google, Inc. All rights reserved.
+// Copyright 2009-2011 Andreas Krennmair. All rights reserved.
+// TLS decoder is a contribution from Jose Selvi
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/google/gopacket"
+)
+
+// TLSType defines the type of data after the TLS Record
+type TLSType uint8
+
+// TLSType known values.
+const (
+	TLSchange_cipher_spec TLSType = 20
+	TLSalert              TLSType = 21
+	TLShandshake          TLSType = 22
+	TLSapplication_data   TLSType = 23
+	TLSunknown            TLSType = 255
+)
+
+func (tt TLSType) String() string {
+	switch tt {
+	default:
+		return "Unknown"
+	case TLSchange_cipher_spec:
+		return "Change Cipher Spec"
+	case TLSalert:
+		return "Alert"
+	case TLShandshake:
+		return "Handshake"
+	case TLSapplication_data:
+		return "Application Data"
+	}
+}
+
+type TLSVersion struct {
```

Comment:
How about:
```go
type TLSVersion uint16

func (tv TLSVersion) Major() uint8 {
  return uint8(tv >> 8)
}
func (tv TLSVersion) Minor() uint8 {
  return uint8(tv)
}
```
This way, you can pull it from the packet with:

`Version = TLSVersion(binary.BigEndian.Uint16(data[x:y]))`